### PR TITLE
feat: add buttons to copy all ids of a slice's/tag's instances

### PR DESCRIPTION
**Pull request by Sparkier**
# Description

fix ZEN-370

### CODE REVIEW

File: backend/zeno_backend/database/select.py [hunk 1]
```diff
@@ -1945,6 +1945,40 @@ def slice_or_tag_table(
...
+    """
+    with Database() as db:
+        if db.cur is None:
+            return []
+
+        filter = sql.SQL("")
+        if filter_sql is not None:
+            filter = sql.SQL("WHERE ") + filter_sql
+
+        final_statement = sql.SQL(" ").join(
```
Comment (cabreraalex) [inline backend/zeno_backend/database/select.py:1969]: Can this just be:

`sql.SQL("SELECT {} FROM {}").format(sql.Identifier(id_column.id), sql.Identifier(uuid)))`

File: backend/zeno_backend/database/select.py [hunk 2]
```diff
@@ -1945,6 +1945,40 @@ def slice_or_tag_table(
...
+    Args:
+        project_uuid (str): project the slice belongs to.
+        filter_sql (sql.Composed | None): filter specified by the slice.
+        id_column (ZenoColumn): identifier of the instance id column in the project.
+
+    Returns:
+        list[str]: list of instance ids that belong to the slice.
+    """
+    with Database() as db:
+        if db.cur is None:
```
Comment (cabreraalex) [inline backend/zeno_backend/database/select.py:1962]: I don't think you need a cursor for this function.

File: backend/zeno_backend/routers/slice.py [hunk 3]
```diff
@@ -114,6 +115,41 @@ def add_slice(project: str, slice: Slice, current_user=Depends(util.auth.claim()
     return id
 
 
+@router.post(
+    "/slice_instance_ids/{slice_id}/{model}", response_model=list[str], tags=["zeno"]
```
Comment (cabreraalex) [inline backend/zeno_backend/routers/slice.py:119]: should be dashes?

File: frontend/src/lib/components/metadata/cells/SliceCell.svelte [hunk 4]
```diff
@@ -183,7 +183,25 @@
...
+									if (idColumn) {
+										zenoClient
+											.getSliceInstanceIds(slice.id, $model === undefined ? null : $model, idColumn)
+											.then((sliceInstanceIds) => {
+												navigator.clipboard.writeText(sliceInstanceIds.toString());
+											});
+									}
+								}}
+							>
+								<Icon style="font-size: 18px;" class="material-icons">delete_outline</Icon>&nbsp;
```
Comment (cabreraalex) [inline frontend/src/lib/components/metadata/cells/SliceCell.svelte:200]: change icon for this and other one

File: frontend/src/lib/components/metadata/cells/TagCell.svelte [hunk 5]
```diff
@@ -201,7 +201,18 @@
 								</button>
 							{/if}
 							<button
-								class="py flex w-20 items-center px-2 hover:bg-grey-lighter"
+								class="py flex w-24 items-center px-2 hover:bg-grey-lighter"
+								on:click={(e) => {
+									e.stopPropagation();
+									showOptions = false;
+									navigator.clipboard.writeText(tag.dataIds.toString());
```
Comment (cabreraalex) [inline frontend/src/lib/components/metadata/cells/TagCell.svelte:208]: I think this should copy with the two brackets `[ ]` around it so you can just directly paste into Python.

**Review by cabreraalex — CHANGES_REQUESTED** [review]

**Comment by Sparkier** [comment]
Thanks. Fixed.

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**